packages/terra-date-picker/src/react-datepicker/week.jsx

```diff
@@ -6,28 +6,97 @@ import * as utils from './date_utils'
 
 export default class Week extends React.Component {
   static propTypes = {
+    /**
+     * Date Value Entered.
+     */
     day: PropTypes.object.isRequired,
+    /**
+     * A callback function for custom day class names.
+     */
     dayClassName: PropTypes.func,
+    /**
+     * Maximum Date value for a date range.
+     */
     endDate: PropTypes.object,
+    /**
+     * Array to store values of time that are disabled to pick .
+     */
     excludeDates: PropTypes.array,
+    /**
+     * A callback function to filter specific dates that can be selected only .
+     */
     filterDate: PropTypes.func,
+    /**
+     * A callback function to format week number .
+     */
     formatWeekNumber: PropTypes.func,
+    /**
+     * Highlight range of dates with custom classes.
+     */
     highlightDates: PropTypes.instanceOf(Map),
+    /**
+     * Show dates only in the given array.
+     */
     includeDates: PropTypes.array,
+    /**
+     * Prop to show inline version of date picker component.
+     */
     inline: PropTypes.bool,
+    /**
+     * Maximum Value of date that can be selected by user.
+     */
     maxDate: PropTypes.object,
+    /**
+     * Minimum Value of date that can be selected by user.
+     */
     minDate: PropTypes.object,
+    /**
+     * Month value picked by user.
+     */
     month: PropTypes.number,
+    /**
+     * A callback function to execute when a date is selected.
+     */
     onDayClick: PropTypes.func,
+    /**
+     * A callback function to execute when day is clicked by mouse.
```

Comment:
i doubt this is invoked on mouse click, more like on mouse hover. Could be "A callback function to execute when mouse cursor hovers over the day"